R/init.R

```diff
@@ -164,6 +164,13 @@ init <- function(data,
   # convert teal.slice::teal_slices to teal::teal_slices
   filter <- as.teal_slices(as.list(filter))
 
+  # Calculate app hash to ensure snapshot compatibility. See ?snapshot. Raw data must be extracted from environments.
+  hashables <- mget(c("data", "modules"))
+  hashables$data <- lapply(hashables$data$get_datanames(), function(dn) {
+    try(hashables$data$get_dataset(dn)$get_raw_data(), silent = TRUE)
+  }) # `try` handles errors from remote data
+  attr(filter, "app_id") <- rlang::hash(hashables)
```

Comment:
There is no raw data in case of connector or DDL. This hash could be calculated after pull (teal_with_splash_srv). If it stays here it could be calculated just on datanames.

Reply:
You're right. But aren't we due to remove `teal_with_splash`?

Reply:
(future) Removal or keeping the `teal_with_splash` shouldn't influence this solution. DDL can't provide this information in the `teal::init` as  shiny session hasn't started yet

Reply:
That's true. I will move it but note that this makes the process a whole lot less clear.

Reply:
So in `teal_with_splash` the data is still not pulled and only a temporary `NULL` is present. I'm not sure how deep this will have to go.

Reply:
Oh, it's even worse. Down in `module_teal` `raw_data()`  is NULL, there aren't even dataset names.

Reply:
Yup, I just noticed it on my refactor branch. This makes sense because data is returned when submit button is clicked, otherwise it is NULL. I think in case of DDL hash could be built on the reproducible code instead?

Reply:
[How's this?](https://github.com/insightsengineering/teal/pull/929/commits/d0e86d7973821473dcc6c308d189cbbad1d8cfd2)?

Reply:
better ;)